### PR TITLE
Add Changelog for v0.5.0 and remove deprecations

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

As we are preparing for v0.5.0 in this PR we take advantage of our first release to remove inherited deprecations across the project. We're listing them as well in the changelog together with the breaking changes introduced after fixing the linting errors. 

## Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used, if not `main`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+- Deprecated `ActionManager.ListTriggers()`
+- Deprecated `ActionManager.ReadVersion()`
+- Deprecated `ActionManager.ListVersions()`
+- Deprecated `ActionManager.ListBindings()`
+- Deprecated `ActionManager.ReadExecution()`
+- Deprecated `ClientRefreshToken.Type`
+- Deprecated `WithFields()`
+- Deprecated `WithoutFields()`
+- Deprecated `TenantFlags.ChangePasswordFlowV1`
```
Comment (willvedd) [inline CHANGELOG.md:27]: These appear to already have been deprecated previously so I think the should refer to them as "API removals".

**Review by willvedd — APPROVED** [review]
Agree with your decision to remove all inherited deprecations. I think we should just reword it to communicate that we removed those deprecations, otherwise sounds like we're just now marking them as deprecated.

**Review by cocojoe — APPROVED** [review]
